KOGITO-7606 Custom dashboards (#231)
kiegroup/kogito-docs: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-custom-dashboard-page.adoc

> = Dashboards in Kogito Serverless Workflow Tools extension

Reviewer: How do you access this page ? Missing into the `nav.adoc` file ?

Author: ok

In Kogito Serverless Workflow Tools, the Dashboards page is used to displays the available dashboard files. The page displays a list of available dashboard and add filters to the list.

.Dashboard page
image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-dashboard-list.png[]

The table on the Dashboards page displays the following details:

> * *Name*: Displays the name of a custom dashboard.
> * *Last Modified*: Displays the time reference indicating when the custom dashboard is last updated.
> 

Reviewer: ```suggestion
* *Name*: displays the name of a custom dashboard.
* *Last Modified*: displays the time reference indicating when the custom dashboard is last updated.

```

> To filter the dashboard based on name, enter a name in the *Filter by name* field and click *Apply filter* button.

Reviewer: screenshot is showing `Filter by dashboard` field

.Example filters applied on the dashboards
image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-filters-in-dashbaords.png[]

> In the dashboard table, the *name* column contains a clickable link, which enables you to navigate to the *Dashboard Details* page.

Reviewer: I would setup 2 subsections to this page:

- Filtering the dashboard(s)
- Accessing a dashboard details

That way we would have a right menu with the direct link to what we want to achieve. wdyt ?

Author: done

> = Dashboards in Kogito Serverless Workflow Tools extension

Reviewer: The page is missing those:
```
:compat-mode!:
// Metadata:
:description: ...
:keywords: kogito, workflow, serverless, Quarkus, Dev UI, dashboards
```

Reviewer: See https://raw.githubusercontent.com/kiegroup/kogito-docs/main/serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-workflow-definition-page.adoc as an example

> In Kogito Serverless Workflow Tools, the Dashboards page is used to displays the available dashboard files. The page displays a list of available dashboard and add filters to the list.

Reviewer 2: ```suggestion
In Kogito Serverless Workflow Tools, the Dashboards page is used to display the available dashboard files. The page displays a list of available dashboards and add filters to the list.
```

.Dashboard page
image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-dashboard-list.png[]

The table on the Dashboards page displays the following details:

> * *Name*: displays the name of a custom dashboard.

Reviewer 2: ```suggestion
* *Name*: the name of a custom dashboard.
```

> * *Last Modified*: displays the time reference indicating when the custom dashboard is last updated.

Reviewer 2: ```suggestion
* *Last Modified*: the time reference indicating when the custom dashboard was last updated.
```

> === Filtering the dashboards

Reviewer: ```suggestion
== Filtering the dashboards
```

> To filter the dashboard based on name, enter a name in the *Filter by dashboard name* field and click *Apply filter* button.

Reviewer 3: ```suggestion
To filter the dashboard based on name, enter a name in the *Filter by dashboard name* field and click the *Apply filter* button.
```

.Example filters applied on the dashboards
image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-filters-in-dashbaords.png[]

> === Accessing a dashboard details

Reviewer: ```suggestion
== Accessing a dashboard details
```

In the dashboard table, the *name* column contains a clickable link, which enables you to navigate to the *Dashboard Details* page.

.Dashboard Details page

> image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-dashboard-details.png[]

Reviewer: Missing the `Found an issue` part. See other pages as the last section

Author: Done

= Dashboards in Kogito Serverless Workflow Tools extension
:compat-mode!:
// Metadata:
:description: Dashboards in Kogito Serverless Workflow Tools extension
:keywords: kogito, workflow, serverless, Quarkus, Dev UI, Dashboards
:dashboard_guide: https://www.dashbuilder.org/docs/#chap-dashbuilder-yaml-guides
:dashboard_editor: https://start.kubesmarts.org/

In Kogito Serverless Workflow Tools, the Dashboards page is used to display the available dashboard files. The page displays a list of available dashboards and add filters to the list.

.Dashboard page
image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-dashboard-list.png[]

The table on the Dashboards page displays the following details:

* *Name*: the name of a custom dashboard.
* *Last Modified*: the time reference indicating when the custom dashboard was last updated.

> == Creating custom dashboard

Reviewer 2: ```suggestion
== Creating a custom dashboard
```

=== Create a custom dashboard file
The {dashboard_guide}[dashboard guide] for creating dashboards and visualizations with YAML. You can run all examples with {dashboard_editor}[Dashbuilder YML Online].

> === Storage path of custom dashboard

Reviewer 2: ```suggestion
=== Storage path of custom dashboards
```

> Default storage path is *dashboards*, user also can use property *quarkus.kogito-runtime-tools.custom.dashboard.folder* to set new custom storage path.

Reviewer 2: ```suggestion
The default storage path for dashboard files is *src/main/resources/dashboards*, but the property *quarkus.kogito-runtime-tools.custom.dashboard.folder* can be used to set a custom storage path.
```

.Default storage path

> image::tooling/quarkus-dev-ui-extension/kogtio-swf-tools-dashboard-storage-path.png[]

Reviewer 2: ```suggestion
image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-dashboard-storage-path.png[]
```
The file should be renamed too.

Author: Done

= Dashboards in Kogito Serverless Workflow Tools extension
:compat-mode!:
// Metadata:
:description: Dashboards in Kogito Serverless Workflow Tools extension
:keywords: kogito, workflow, serverless, Quarkus, Dev UI, Dashboards
:dashboard_guide: https://www.dashbuilder.org/docs/#chap-dashbuilder-yaml-guides
:dashboard_editor: https://start.kubesmarts.org/

In Kogito Serverless Workflow Tools, the Dashboards page is used to display the available dashboard files. The page displays a list of available dashboards and add filters to the list.

.Dashboard page
image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-dashboard-list.png[]

The table on the Dashboards page displays the following details:

* *Name*: the name of a custom dashboard.
* *Last Modified*: the time reference indicating when the custom dashboard was last updated.

== Creating a custom dashboard

=== Create a custom dashboard file

> The {dashboard_guide}[dashboard guide] for creating dashboards and visualizations with YAML. You can run all examples with {dashboard_editor}[Dashbuilder YML Online].

Reviewer 4: This sentence is missing something, it sounds more like a description.
`The {dashboard_guide}[dashboard guide] for creating dashboards and visualizations with YAML.`

Reviewer 4: This still holds. What is the purpose of this sentence? `The {dashboard_guide}[dashboard guide] for creating dashboards and visualizations with YAML.` Does it describe any picture or something similar?

Author: Custom dashboard has to create a dashboard file first, so we need let user know how to create it. 
Create dashboard does not belong to rumtime-tools, so I think the external reference is better.

Reviewer 4: I know, just the sentence is weirdly written to me :) It preferably could be `See the dashboard guide for creating dashboards and visualizations with YAML.` or similar.

Author: Done

> The {dashboard_guide}[dashboard guide] for creating dashboards and visualizations with YAML. You can run all examples with {dashboard_editor}[Dashbuilder YML Online].

Reviewer 4: We should probably unify YAML/YML, so mostlikely YAML.
```suggestion
The {dashboard_guide}[dashboard guide] for creating dashboards and visualizations with YAML. You can run all examples with {dashboard_editor}[Dashbuilder YAML Online].
```

Author: Done
